### PR TITLE
Cumulative Update (beta)

### PR DESCRIPTION
**Pull request by akvlad**
# Updates description

## Days / weeks support in the duration brackets https://github.com/metrico/qryn/issues/261
`[1m/h/s]` - these square brackets support days (d) and weeks (w) like `count_over_time({....}[1w])`

## Full compression support https://github.com/metrico/qryn/issues/239

Compressed requests and responses are fully supported as described here https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding

## position -> like / notLike as requested in https://github.com/metrico/qryn/issues/213

Internal update of clickhouse queries. Now they use `LIKE '%...%'` to provide `|=` operator

## Configurable primary key as requested in https://github.com/metrico/qryn/issues/212

Please specify the `ADVANCED_SAMPLES_ORDERING` env var with the raw clickhouse ORDER BY (...) experssion.
If you want use fingerprint, please specify `ADVANCED_SAMPLES_ORDERING=fingerprint,timestamp_ns` . 
This env variable cannot be changed for the same database.
If you specified this configuration, no samples primary key updates will be applied for the database in the future.
Please use it with care. 

----

Reference PRs:
- https://github.com/metrico/qryn/pull/279
- https://github.com/metrico/qryn/pull/278
- https://github.com/metrico/qryn/pull/235
- https://github.com/metrico/qryn/pull/216

### CODE REVIEW

**Comment by akvlad** [comment]
@Dletta we have a lot of updates to add to the documentation before merge.

**Comment by akvlad** [comment]
# Updates description

## Days / weeks support in the duration brackets
`[1m/h/s]` - these square brackets support days (d) and weeks (w) like `count_over_time({....}[1w])`

## Full compression support

Compressed requests and responses are fully supported as described here https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding

## position -> like / notLike

Internal update of clickhouse queries. Now they use `LIKE '%...%'` to provide `|=` operator

## Configurable primary key as requested in https://github.com/metrico/qryn/issues/212

Please specify the `ADVANCED_SAMPLES_ORDERING` env var with the raw clickhouse ORDER BY (...) experssion.
If you want use fingerprint, please specify `ADVANCED_SAMPLES_ORDERING=fingerprint,timestamp_ns` . 
This env variable cannot be changed for the same database.
If you specified this configuration, no samples primary key updates will be applied for the database in the future.
Please use it with care.

**Comment by akvlad** [comment]
@lmangani please reflect it inside the changelog.

**Comment by Dletta** [comment]
I have tested items per below checklist, you have pointed out as changed, plus checking against our qryn development dashboard. I can confirm each item marked functions as expected at this time:

- [X] days/weeks in range queries
- [X] Full compressions in and out
- [X] position -> like/not_like
- [ ] Custom Order By setting (I did not test this item at this time)

File: parser/transpiler.js [hunk 1]
```diff
@@ -169,6 +169,7 @@ module.exports.transpile = (request) => {
   setQueryParam(query, sharedParamNames.to, end + '000000')
   setQueryParam(query, 'isMatrix', query.ctx.matrix)
   logger.debug(query.toString())
+  console.log(query.toString())
```
Comment (Dletta) [inline parser/transpiler.js:172]: Is this left from debugging?

**Review by Dletta — APPROVED** [review]

**Review by Dletta — APPROVED** [review]

**Pull request merged**